phoenix-core/src/main/java/org/apache/phoenix/monitoring/connqueryservice/PhoenixConnectionQueryServiceMetrics.java

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.monitoring.connqueryservice;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.phoenix.monitoring.MetricType;
+import org.apache.phoenix.monitoring.PhoenixConnectionProfileMetric;
+import org.apache.phoenix.monitoring.PhoenixConnectionProfileMetricImpl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.phoenix.monitoring.MetricType.OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER;
+import static org.apache.phoenix.monitoring.MetricType.OPEN_PHOENIX_CONNECTIONS_COUNTER;
+import static org.apache.phoenix.monitoring.MetricType.PHOENIX_CONNECTIONS_THROTTLED_COUNTER;
+
+public class PhoenixConnectionQueryServiceMetrics {
+    public enum ConnectionQueryServiceMetrics {
+        CONNECTION_QUERY_SERVICE_OPEN_PHOENIX_CONNECTIONS_COUNTER(OPEN_PHOENIX_CONNECTIONS_COUNTER),
+        CONNECTION_QUERY_SERVICE_OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER(OPEN_INTERNAL_PHOENIX_CONNECTIONS_COUNTER),
+        CONNECTION_QUERY_SERVICE_PHOENIX_CONNECTIONS_THROTTLED_COUNTER(PHOENIX_CONNECTIONS_THROTTLED_COUNTER);
+
+        private MetricType metricType;
+        private PhoenixConnectionProfileMetric metric;
+
+        ConnectionQueryServiceMetrics(MetricType metricType) {
+            this.metricType = metricType;
+        }
+    }
+
+    private final String connectionQueryServiceName;
+    private Map<MetricType, PhoenixConnectionProfileMetric> metricRegister;
+    private ConnectionQueryServiceMetricsHistograms connectionQueryServiceMetricsHistograms;
+
+    public PhoenixConnectionQueryServiceMetrics(final String connectionQueryServiceName,
+            Configuration conf) {
+        this.connectionQueryServiceName = connectionQueryServiceName;
+        metricRegister = new HashMap<>();
+        for (ConnectionQueryServiceMetrics connectionQueryServiceMetric
+                : ConnectionQueryServiceMetrics.values()) {
+            connectionQueryServiceMetric.metric =
+                    new PhoenixConnectionProfileMetricImpl(connectionQueryServiceMetric.metricType);
+            metricRegister.put(connectionQueryServiceMetric.metricType,
+                    connectionQueryServiceMetric.metric);
+        }
+        connectionQueryServiceMetricsHistograms =
+                new ConnectionQueryServiceMetricsHistograms(connectionQueryServiceName, conf);
+    }
+
+    /**
+     * This function is used to update the value of Metric
+     * In case of counter val will be passed as 1.
+     *
+     * @param type metric type
+     * @param val update value. In case of counters, this will be 1
+     */
+    public void changeMetricValue(MetricType type, long val) {
```

Comment:
can this be setMetricValue or updateMetricValue?

Reply:
done